transport/graphsync/graphsync.go

```diff
@@ -635,12 +660,23 @@ func (t *Transport) gsIncomingResponseHook(p peer.ID, response graphsync.Respons
 	if err != nil {
 		hookActions.TerminateWithError(err)
 	}
+
+	// In a case where the transfer sends blocks immediately this extension may contain both a
+	// response message and a revalidation request so we trigger OnResponseReceived again for this
+	// specific extension name
+	msg, err := extension.GetTransferData(response, []graphsync.ExtensionName{extension.ExtensionOutgoingBlock1_1})
```

Comment:
Nit: you should use processExtensions here with the list above. We can just ignore ResponseMessage I think for now?